test/cache_test.go

```diff
@@ -346,3 +355,84 @@ func testAutoGobuildCache(ctx context.Context, t *testing.T, cfg targetConfig) {
 		solveT(ctx, t, client, sr)
 	})
 }
+
+func testAutoCargobuildCache(ctx context.Context, t *testing.T, cfg targetConfig) {
+	ctx = startTestSpan(ctx, t)
+
+	specWithCommand := func(cmd string) *dalec.Spec {
+		spec := newSimpleSpec()
+		spec.Dependencies = &dalec.PackageDependencies{
+			Build: make(map[string]dalec.PackageConstraints),
+		}
+
+		// Handle space-separated package lists from PackageOverrides
+		rustPackages := cfg.GetPackage("rust")
+		for _, pkg := range strings.Fields(rustPackages) {
+			spec.Dependencies.Build[pkg] = dalec.PackageConstraints{}
+		}
+
+		spec.Build.Steps = append(spec.Build.Steps, dalec.BuildStep{
+			Command: cmd,
+		})
+		return spec
+	}
+
+	testEnv.RunTest(ctx, t, func(ctx context.Context, client gwclient.Client) {
+		// Test that cargo cache is NOT automatically enabled (it's opt-in only now)
```

Comment:
Comment is not neccessary, or could be updated to something along the lines of `// test no cache`